src/createAction.js

```diff
@@ -10,35 +9,30 @@ export default function createAction(type, payloadCreator = identity, metaCreato
     'Expected payloadCreator to be a function, undefined or null'
   );
 
-  const finalPayloadCreator = isNull(payloadCreator)
+  const finalPayloadCreator = payloadCreator === null || payloadCreator === identity
```

Comment:
1. No reason. I just figured that the purpose of `isNull` is to be a predicate to functions like `filter`, so there's no reason to have the function call overhead here.
2. The `identity` check means that if the `payloadCreator` is `identity`, we can skip the `instanceof Error` check every time it's called, and don't have to use the latter version of the `finalPayloadCreator`.